testcases/elichika_tests/canonicalizer/Return_test.py

```diff
@@ -0,0 +1,265 @@
+import unittest
+import ast, gast
+from chainer_compiler.elichika.parser import canonicalizer
+from chainer_compiler.elichika.parser import utils
+from chainer_compiler.elichika.testtools import compare_ast, assert_semantically_equals
+
+
+class Return(unittest.TestCase):
+    def setUp(self):
+        self.canonicalizer = canonicalizer.Canonicalizer(use_illegal_identifier=False)
+
+    def test_return(self):
```

Comment:
The canonicalizer modified it to match:
```python
    def test_return_noop(self):
        orig_code = utils.clip_head("""
        def func(a, b):
            return a + b

        value = func(3, 39)
        """)
        target_code = utils.clip_head("""
        def func(a, b):
            # initializing return flags and return value assignment variable.
            returned_value = None
            returned_0 = False
            # replacing return with boolean flag and assignment. 
            returned_0 = True
            returned_value = a + b
            # returning the assigned value.
            return returned_value

        value = func(3, 39)
        """)
        orig_ast = gast.ast_to_gast(ast.parse(orig_code))
        target_ast = gast.ast_to_gast(ast.parse(target_code))
        converted_ast = self.canonicalizer.visit(orig_ast)

        assert_semantically_equals(orig_code, target_code, ['value'])
        assert compare_ast(converted_ast, target_ast)
        assert compare_ast(target_ast, converted_ast)
```

Reply:
It actually helps in handling the following case:
```python
orig_code = utils.clip_head("""
        def func(a, b):
            if a > b:
                 return a - b
            else:
                 return b - a

        value = func(3, 39)
        """)
```
In this case, the canonicalizer replaces all the returns with assignment and there is a singular `return returned_value` statement in the last. The variable `returned_value` would store either `a-b` or `b-a` based on the branch encountered. Because of the recursive implementation, even if there is just one return statement, it is replaced by an assignment, and a new return statement is added in the end.

I can add a special edge case to not do this when there is a singular return statement, but do you think it is a good idea?

Reply:
Ah, it sounds like the current behavior is reasonable. No, I don't think we need to handle the single return case specially. Thanks for you explanation. Could you just add `test_return_noop` case with the expected `target_code` with the current canonicalizer?

Reply:
Yes, Sure. Will do that.